pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -386,27 +386,33 @@ func DeleteIPsecEndpoint(remote *net.IPNet) {
 
 func isXfrmPolicyCilium(policy netlink.XfrmPolicy) bool {
 	if policy.Mark == nil {
+		// Check if its our fwd rule, we don't have a mark
+		// on this rule so use priority.
+		if policy.Dir == netlink.XFRM_DIR_FWD &&
+			policy.Priority == linux_defaults.IPsecFwdPriority {
+			return true
+		}
 		return false
 	}
-	if policy.Mark.Mask != linux_defaults.RouteMarkMask {
-		return false
+
+	if (policy.Mark.Value & linux_defaults.RouteMarkDecrypt) != 0 {
+		return true
 	}
-	if policy.Mark.Value == linux_defaults.RouteMarkDecrypt ||
-		policy.Mark.Value == linux_defaults.RouteMarkEncrypt {
+	if (policy.Mark.Value & linux_defaults.RouteMarkEncrypt) != 0 {
```

Comment:
nit: These two if statements could be combined (it took me a moment to spot the difference between them):

```go
if policy.Mark.Value & (linux_defaults.RouteMarkDecrypt | linux_defaults.RouteMarkEncrypt) != 0 {
    return true
}
```

Reply:
I'm not sure this is a great idea. The mark bits are not unique here, we're treating the `0x0F00` bits as a 4-bit value (to extract up to 15 unique values out of the bits rather than 4). We don't want to see whether bits `0x0F00` are set (which is what `0xD | 0xE` results in), we want to know the rule uses one of two specific values within that set of four bits.

Reply:
I think I agree with @joestringer here. I'm leaning towards being overly verbose.